cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java

```diff
@@ -370,11 +370,20 @@ public void setMSK_SLIDE_ID(String mskSlideID) {
         this.mskSlideID = mskSlideID;
     }
 
-    private String resolveSampleType(Integer isMetastasis) {
-        if (isMetastasis != null)
-            return isMetastasis == 0 ? "Primary" : "Metastasis";
+    public String resolveSampleType(Integer isMetastasis) {
+        if (isMetastasis != null) {
+            switch(isMetastasis) {
+                case 0:
+                    return "Primary";
+                case 1:
+                    return "Metastasis";
+                case 127:
```

Comment:
This is different from the description on the card. Local Recurrence had code of 2, and Unknown had a code of 127. Theoretically any other code coming in is an unexpected condition (maybe an exception or a reason to substitute NA)?